syntax = "proto3";

package GrpcApiPractice;

import "Models/Invoice.proto";
import "Models/ServiceStatus.proto";
import "google/protobuf/empty.proto";

service InvoiceService
{
    rpc CreateInvoice(GrpcApiPractice.Invoice) returns (GrpcApiPractice.ServiceStatus) { }
    rpc ListInvoice(google.protobuf.Empty) returns (ListInvoiceResponse) { }
    rpc ListInvoiceById (InvoiceByIdRequest) returns (ListInvoiceResponse) { }
}

message InvoiceByIdRequest
{
    int64 invoice_id = 1;
}

message ListInvoiceResponse
{
    repeated GrpcApiPractice.Invoice invoice = 1;
}